pgmpy/metrics/metrics.py

```diff
@@ -425,18 +429,40 @@ def SHD(true_model, est_model):
     >>> dag1 = BayesianNetwork([(1, 2), (2, 3)])
     >>> dag2 = BayesianNetwork([(2, 1), (2, 3)])
     >>> SHD(dag1, dag2)
-    2
+    1
     """
     if set(true_model.nodes()) != set(est_model.nodes()):
         raise ValueError("The graphs must have the same number of nodes.")
 
+    nodes_list = true_model.nodes()
+
     dag_true = nx.DiGraph(true_model.edges())
-    dag_true.add_nodes_from(true_model.nodes())
-    adj_mat_true = nx.adjacency_matrix(dag_true).todense()
+    adj_mat_true = nx.adjacency_matrix(dag_true, nodelist=nodes_list).todense()
 
     dag_est = nx.DiGraph(est_model.edges())
-    dag_est.add_nodes_from(est_model.nodes())
-    adj_mat_est = nx.adjacency_matrix(dag_est).todense()
+    adj_mat_est = nx.adjacency_matrix(dag_est, nodelist=nodes_list).todense()
+
+    shd = 0
+
+    m1 = np.array(adj_mat_true)
+    m2 = np.array(adj_mat_est)
+
+    s1 = m1 + m1.T
+    s2 = m2 + m2.T
+
+    s1[s1 == 2] = 1
+    s2[s2 == 2] = 1
```

Comment:
Yep, I think this would only be true when two nodes have edges pointing to each other, which would go against the definition of a DAG. I guess the R implementation is for a general graph.